constrained-voucher.mkd

```diff
@@ -510,6 +512,44 @@ Any POST request to the Registrar with resource /est/vs or /est/es returns a 2.0
 
 # Raw Public Key Use Considerations {#rpk-considerations}
 
+This section explains changes that can be done to reduce how many bytes that are sent as part of the certificates.
+The use of a raw public key in the pinning process can significantly reduce the number of bytes and round trips, but it comes with a few significant operational limitations.
+
+## The Registrar Trust Anchor
+
+When the Pledge first connects to the Registrar, the connection to the Registrar is provisional, as explained in section 5.6.2 of {{I-D.ietf-anima-bootstrapping-keyinfra}}.
+The Registrar provides its public key in a TLSServerCertificate, and the Pledge uses that to validate that integrity of the (D)TLS connection, but it does not validate the identity of
+the provided certificate.
+
+As the TLSServer Certificate object is never verified directly by the pledge, sending it can be considered superfluous.
+Instead of using a (TLSServer)Certificate of type X509 (see section 4.4.2 of {{RFC8446}}),
+a RawPublicKey object is used.
+
+A Registrar operating in a mixed environment can determine whether to send a Certificate or a Raw Public key is determined by the pledge including the server\_certificate\_type of RawPublicKey.
+This is shown in section 5 of {{RFC7250}}.
+
+The Pledge continues to send a client\_certificate\_type of X509.
+
+## The Voucher Request
+
+The Pledge puts the Registrar's public key into the proximity-registrar-pubk
+(PRSPKI) field of the voucher-request.
+(The proximity-registrar-pubk-sha256 can also be used if the 32-bytes of hash turned out to be smaller than a typical ECDSA key.)
+
+As the format of the pubk field is identical to the TLS Certificate RawPublicKey, no manipulation at all is needed to insert this into a voucher-request.
+
+## The Voucher Reply
+
+A returned voucher will have a pinned-domain-subk field with the identical key as was found in the proximity field above, as well as in the TLS connection.
+Validation of this is what takes the connection out of the provisional state (see {{I-D.ietf-anima-bootstrapping-keyinfra}} section 5.6.2).
+
+The voucher needs to be validated first.
+The Pledge needs to have a public key to validate the signature from the MASA on the voucher.
+
+The MASA signing key is based upon a PKI (see {{I-D.richardson-anima-masa-considerations}} section 2.3) , then a certificate chain may need to be included with the voucher in order that the pledge is able to validate the signature.
```

Comment:
I don't see how lines 549-552 fit in the current section. This is a general consideration. If the MASA signs with something that's not pre-installed in the Pledge, it will always need to send additional certs such that the Pledge can chain from signing identity to the MASA CA that is pre-installed. Better move to another part of the draft!